dcompose/dcompose.go

```diff
@@ -207,18 +209,27 @@ func (j *JobCompose) InitFromJob(job *model.Job, cfg *viper.Viper, workingdir st
 	}
 
 	// Add the final output job
+	excludesPath := path.Join(workingdir, UploadExcludesFilename)
+	excludesMount := path.Join(CONFIGDIR, UploadExcludesFilename)
+
+	uploadOutputsCmd := job.FinalOutputArguments()
+
+	uploadOutputsCmd = append(uploadOutputsCmd, "--exclude")
+	uploadOutputsCmd = append(uploadOutputsCmd, excludesMount)
+
 	j.Services["upload_outputs"] = &Service{
 		CapAdd:  []string{"IPC_LOCK"},
 		Image:   porklockImageName,
-		Command: job.FinalOutputArguments(),
+		Command: uploadOutputsCmd,
 		Environment: map[string]string{
 			"VAULT_ADDR":  "${VAULT_ADDR}",
 			"VAULT_TOKEN": "${VAULT_TOKEN}",
 			"JOB_UUID":    job.InvocationID,
 		},
 		WorkingDir: WORKDIR,
 		Volumes: []string{
-			fmt.Sprintf("%s:%s:%s", job.InvocationID, WORKDIR, "rw"),
+			strings.Join([]string{job.InvocationID, WORKDIR, "rw"}, ":"),
+			strings.Join([]string{excludesPath, excludesMount, "ro"}, ":"),
```

Comment:
You're copying this file into the `logs` directory, which is inside the `WORKDIR` anyway, right? (certainly if we're uploading it to iRODS it's in the place where the output porklock could already see it). It seems like it'd be easier to use it directly from there. I suppose that does technically mean a tool could edit the contents of the file, though, if that's what you're avoiding?

Reply:
Yes, I've come across a tool that wanted to create its own `log` directory in the working directory, and it cleared out all files under that directory at the start of its run. So I don't want to rely on anything in the `log` dir after the tool runs.

So for now I'm leaving the file we need to use outside the working directory to prevent any corruption by the tool. I've also wondered before if we could keep all our logs outside the tool's working directory as well, but that might require changing the way outputs come back to the user's results folder (so the tool's outputs are contained in a separate folder), which may be considered a big change to our users?

Reply:
Hm, possibly. We've changed a lot of stuff around like that before though (the job system rewrite changed around a lot of filenames and structures) so I think we'd possibly be fine doing that. Anyway, I think you're right about it for now.